Display accurate build status at any point

robertboloc opened the pull request:
Currently, after pushing a new commit if a user asks for the status image, the displayed status will be either failing or passing, based on the previous commit, even if the current build has not completed. 

If the status image is linked somewhere (we use a chrome plugin to embed this image automatically on all our Bitbucket pull requests) most of the time the build is shown as passing, until the actual build completes, and the image is updated.

This causes issues as the build takes a while to complete, and developers have to manually check the PHPCI interface for any running builds before actually trusting the image. I think this defeats the purpose of the badge.

This pull request adds 2 more statuses for the badge (not the build):
- unknown (build is created but is not being executed yet), light grey color
- running (build is being executed by the daemon or cron task), orange color

This displays the actual state of a branch at any point, avoiding to have to manually check for any running builds

dancryer commented:
Happy to merge this - but I'd prefer it if the badge used the same set of statuses we use in PHPCI itself... Pending (blue), Running (yellow), Success (green), Failed (red)

robertboloc commented:
Thanks for the feedback :) I have updated the code to use the same statuses and colours as the rest of the application.

I have also rebased the branch to the latest master.

mikebronner commented:
:+1: after rebasing again :)

robertboloc commented:
Rebased :+1:

mikebronner commented:
:+1:
